code/game/atoms_movable.dm

```diff
@@ -719,3 +719,46 @@
 
 /atom/movable/proc/can_apply_inertia()
 	return (!src.anchored && !(src.pulledby && src.pulledby.Adjacent(src)))
+
+/atom/movable/proc/send_to_future(var/duration)
+	var/init_invisibility = invisibility
+	var/init_invuln = flags & INVULNERABLE
+	var/init_density = density
+	var/init_anchored = anchored
+	var/init_timeless = flags & TIMELESS
+	var/init_blinded
+	var/mob/M
+	if(ismob(src))
```

Comment:
No src type checks. Specific behavior like this should go on subtypes.

If you need to modify the spawn, you can give the proc `set waitfor = 0` on the root level and replace the spawn with a sleep so to *callers* it's effectively the same, but anything overriding it can wrap code around the `..()` without issues.